packages/angular/cli/models/schematic-command.ts

```diff
@@ -256,7 +256,7 @@ export abstract class SchematicCommand<
       registry: new schema.CoreSchemaRegistry(formats.standardFormats),
       resolvePaths: !!this.workspace.configFile
         // Workspace
-        ? [process.cwd(), this.workspace.root, __dirname]
+        ? [__dirname, process.cwd(), this.workspace.root]
```

Comment:
This will cause very different schematic resolution especially in monorepos.  The location of invocation should take precedence locally.
The root cause of the referenced issue is really package hoisting combined with a dependency using a much older version of the built-in Angular schematics.  Unfortunately we cannot fix that so an alternative may be to special case the built-in `@schematics/angular` package.  This would break users intentionally  installing a different version of the package but is probably the least invasive fix.